### PR TITLE
Fix/shipping profile

### PR DESCRIPTION
**Pull request by Skieter**

### CODE REVIEW

**Comment by HenkSwardt** [comment]
https://github.com/particl/particl-desktop/issues/886

File: src/app/market/buy/buy.component.ts [hunk 1]
```diff
@@ -20,14 +21,26 @@ import { CountryList } from 'app/core/market/api/listing/countrylist.model';
   templateUrl: './buy.component.html',
   styleUrls: ['./buy.component.scss']
 })
-export class BuyComponent implements OnInit {
+export class BuyComponent implements OnInit, DoCheck {
 
+  @ViewChild('stepper') stepper: MatStepper;
+  private modelFirstName: any;
```
Comment (anandsinghparihar) [inline src/app/market/buy/buy.component.ts]: Please create single model  for form data. ? 
something like `ShippingDetail`

File: src/app/market/buy/buy.component.html [hunk 2]
```diff
@@ -151,48 +151,48 @@
...
+                  <input matInput placeholder="First name" [(ngModel)]="modelFirstName" [ngModelOptions]="{standalone: true}" (keyup)="onKeyFirstName(tagFirstName.value)" #tagFirstName>
                 </mat-form-field>
                 <mat-form-field class="full-width" fxFlex>
-                  <input matInput placeholder="Last name">
+                  <input matInput placeholder="Last name" [(ngModel)]="modelLastName" [ngModelOptions]="{standalone: true}" (keyup)="onKeyLastName(tagLastName.value)" #tagLastName>
                 </mat-form-field>
               </div>
               <mat-form-field class="full-width">
-                <input matInput formControlName="addressLine1" placeholder="Address line 1">
+                <input matInput formControlName="addressLine1" placeholder="Address line 1" [(ngModel)]="modelAddressLine1" (keyup)="onKeyAddressLine1(tagAddressLine1.value)"  #tagAddressLine1>
```
Comment (anandsinghparihar) [inline src/app/market/buy/buy.component.html]: May I know why you use keyup and # reference

File: src/app/market/buy/buy.component.ts [hunk 3]
```diff
@@ -200,6 +236,49 @@ export class BuyComponent implements OnInit {
     }
   }
 
+  static ShippingDetails = class {
```
Comment (anandsinghparihar) [inline src/app/market/buy/buy.component.ts]: move it to separate file `shipping-details.model.ts`

File: src/app/market/buy/buy.component.ts [hunk 4]
```diff
@@ -200,6 +236,49 @@ export class BuyComponent implements OnInit {
...
+    static lastName: string;
+    static addressLine1: string;
+    static addressLine2: string;
+    static city: string;
+    static zip: string;
+    static state: string;
+    static country: string;
+  }
+
+  onKeyFirstName(value: string) {
```
Comment (anandsinghparihar) [inline src/app/market/buy/buy.component.ts]: if we using 'ngModel` then why we need this, ngModel will automatically bind value

File: src/app/market/buy/buy.component.ts [hunk 5]
```diff
@@ -155,7 +171,27 @@ export class BuyComponent implements OnInit {
...
       });
-    })
+    });
+  }
+
+  ngDoCheck(): void {
+    try {  
+      this.stepper.selectedIndex = BuyComponent.stepperIndex;   
+      
+      this.modelFirstName = BuyComponent.ShippingDetails.firstName;
```
Comment (anandsinghparihar) [inline src/app/market/buy/buy.component.ts]: I am not sure why you manually assign a value to model?

**Comment by HenkSwardt** [comment]
As discussed with Kirti, He will apply this change to #911 as part of one PR.

**Comment by HenkSwardt** [comment]
The change has been merged to #911, closing PR as it will not be needed.

**Pull request closed without merging**